### ISSUE TITLE
Classifier Loss Function computed from predict_proba()

### ISSUE THREAD

**Issue by adodge** [hyperopt/hyperopt-sklearn#108]
Hello!  I have an interest in doing hyperparameter optimization using something like AUC or cross entropy as a loss function.  I believe the current implementation doesn't allow this, because the loss_fn for classifiers is given the (boolean) output of model.predict(), and these losses are a function of the predicted probability.  I have a workaround that I've been using, which is to wrap my classifier in something like this:

```
class ProbabilityRegressor(BaseEstimator, RegressorMixin):
    """
    A bodge so we can use loss functions that care about the probability not
    the prediction for classifiers.

    Wraps a classifier and calls .predict_proba() when .predict() is called.
    """
    def __init__(self, estimator):
        self.estimator = estimator
    def predict(self, X):
        return self.estimator.predict_proba(X)
    def fit(self,X,y=None):
        self.estimator.fit(X,y)
        return self
    def fit_predict(self,X,y=None):
        self.fit(X,y)
        return self.predict(X)
```

This lets me pretend my classifier is a regressor and I can then use the loss function I want.

This feels like quite a hack, though.  I have a patch where I've added this functionality to the HyperoptEstimator in hopefully a non-intrusive way..  I add a new parameter `continuous_loss_fn` which, when True, causes the `loss_fn` to be called using the output of predict_proba.

For example:

```
from hpsklearn import HyperoptEstimator, sgd
from sklearn.metrics import log_loss
import numpy as np

# Generate some random data
X = np.hstack([
    np.vstack([
        np.random.normal(0,1,size=(1000,10)),
        np.random.normal(1,1,size=(1000,10)),
    ]),
    np.random.normal(0,1,size=(2000,10)),
])
y = np.zeros(2000)
y[:1000] = 1

def loss_function(targ, pred):
    # hyperopt_estimator flattens the prediction when saving it.  This also
    # affects multilabel classification.
    pred = pred.reshape( (-1, 2) )
    return log_loss(targ, pred[:,1])

# Try to fit an SGD model using log_loss as the loss function
cls = HyperoptEstimator(
    classifier=sgd('sgd', loss='log'),
    preprocessing=[],
    loss_fn = loss_function,
    continuous_loss_fn=True,
)
cls.fit(X,y,cv_shuffle=True)
```
